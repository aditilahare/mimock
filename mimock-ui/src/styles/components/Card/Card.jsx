import React from 'react';
import PropTypes from 'prop-types';
import { CardContainer } from './Card.style';

function Card({
	children,
	dataTestid,
	padding,
	border,
	background,
	width,
	margin,
}) {
	return (
		<CardContainer
			data-testid={dataTestid}
			$background={background}
			$border={border}
			$padding={padding}
			$width={width}
			$margin={margin}
		>
			{children}
		</CardContainer>
	);
}

Card.propTypes = {
	children: PropTypes.any.isRequired,
	padding: PropTypes.string,
	border: PropTypes.string,
	background: PropTypes.string,
	width: PropTypes.string,
	margin: PropTypes.string,
	dataTestid: PropTypes.string,
};

export default Card;
