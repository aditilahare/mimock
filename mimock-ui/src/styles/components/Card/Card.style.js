import tw from 'tailwind-styled-components';

export const CardContainer = tw.div`
    w-fit
    rounded-md
    p-8
    shadow-lg
    m-10
    ${(props) => props.$color}
	${(props) => props.$background}
	${(props) => props.$width}
	${(props) => props.$padding}
    ${(props) => props.$margin}
	${(props) => props.$border}
`;
