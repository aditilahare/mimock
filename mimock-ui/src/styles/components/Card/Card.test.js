import React from 'react';
import { render } from '@testing-library/react';
import Card from './Card';

describe('Card', () => {
	let tree;

	it('should render card', async () => {
		tree = await render(
			<Card
				dataTestid='card-test'
				$background={'bg-red-100'}
				$border={'border-2 border-red-500'}
			>
				Hello
			</Card>
		);

		const { getByTestId, container } = tree;

		expect(getByTestId('card-test')).toBeInTheDocument();
		expect(container).toMatchSnapshot();
	});
});
