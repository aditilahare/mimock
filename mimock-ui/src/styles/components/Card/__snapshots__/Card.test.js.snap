// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should render card 1`] = `
<div>
  <div
    class="w-fit rounded-md p-8 shadow-lg m-10"
    data-testid="card-test"
  >
    Hello
  </div>
</div>
`;
